flink-kubernetes/src/test/java/org/apache/flink/kubernetes/highavailability/KubernetesLeaderElectionDriverTest.java

```diff
@@ -118,7 +122,9 @@ public void testWriteLeaderInformationWhenConfigMapNotExist() throws Exception {
 					final String errorMsg = "Could not write leader information since ConfigMap "
 						+ LEADER_CONFIGMAP_NAME + " does not exist.";
 					assertThat(electionEventHandler.getError(), is(notNullValue()));
-					assertThat(electionEventHandler.getError().getMessage(), containsString(errorMsg));
+					assertThat(
+						electionEventHandler.getError(),
+						FlinkMatchers.containsCause(new KubernetesException(errorMsg)));
```

Comment:
Maybe it makes sense to add a new matcher which only checks the message: `FlinkMatchers.containsMessage(errorMsg)`. That way the test would not depend on the actually thrown exception type (here `KubernetesException`).